lib/waylon/jenkins/job/rest.rb

```diff
@@ -26,13 +26,13 @@ def status
 
         def est_duration
           # estimatedDuration is returned in ms; here we convert it to seconds
-          client.api_get_request("/job/#{@name}/lastBuild", nil, '/api/json?depth=1&tree=estimatedDuration')['estimatedDuration'] / 1000
+          @est_duration ||= client.api_get_request("/job/#{URI.escape @name}/lastBuild", nil, '/api/json?depth=1&tree=estimatedDuration')['estimatedDuration'] / 1000
```

Comment:
If jenkins reports a wildly inaccurate estimated duration, then we won't "correct" that value over time, but honestly I've never used the est duration functionality in waylon, and I'd rather see the perf benefit of caching the initial value.